src/Chapter4.hs

```diff
@@ -472,10 +478,11 @@ Implement the Applicative instance for our 'Secret' data type from before.
 -}
 instance Applicative (Secret e) where
     pure :: a -> Secret e a
-    pure = error "pure Secret: Not implemented!"
+    pure = Reward
 
     (<*>) :: Secret e (a -> b) -> Secret e a -> Secret e b
-    (<*>) = error "(<*>) Secret: Not implemented!"
+    (<*>) (Trap e) _ = Trap e
+    (<*>) (Reward f) a = Reward f a
```

Comment:
That's right!

With operators, you can specify the body in the infix form as well:

```suggestion
    Trap e <*> _ = Trap e
    Reward f <*> a = Reward f a
```

This would do the same 👌🏼